test: stale assertion in test_deploy_cerberus_secrets — checks for old error text the code no longer prints

## Plain-English summary

A test is checking that an error message contains a specific phrase. Someone later **reworded that error message in the code** but **didn't update the test**. So the test now looks for words that the program no longer says, and fails — even though the program is behaving perfectly correctly. **The code is right; the test is out of date.**

This is a test-only failure. Nothing about the actual `deploy_cerberus_secrets.py` behavior is broken.

## What the test does

`tests/test_deploy_cerberus_secrets.py::test_main_returns_1_when_no_pem_and_no_dry_run` calls the tool with no arguments (no `.pem` file, no `--dry-run`). That's an invalid way to run it, so the tool should refuse and print an error. The test checks two things:

1. The tool returns exit code `1` (failure). ✅ **This still passes.**
2. The error text printed to stderr contains the phrase `"pem_path is required"`. ❌ **This fails.**

## Why it fails

The error message in the code was reworded — almost certainly when the `--cerberus-pem-gpg` (encrypted-key) option was added (#1254). The message used to say something like *"pem_path is required"*. It now says:

```
ERROR: pem_path (plaintext) or --cerberus-pem-gpg PATH (encrypted) is required unless --dry-run is set.
```

(See `tools/deploy_cerberus_secrets.py:314-318`.)

Notice the new message **no longer contains the exact substring `"pem_path is required"`** — the words `pem_path` and `is required` are both there, but separated by `(plaintext) or --cerberus-pem-gpg PATH (encrypted)`. The test does a literal substring match, so it doesn't find the phrase and fails. The tool itself did exactly the right thing: returned exit 1 and printed a clear, *more informative* error.

## Reproduction (deterministic — not environment-specific)

```
$ poetry run pytest "tests/test_deploy_cerberus_secrets.py::test_main_returns_1_when_no_pem_and_no_dry_run" -q
E   AssertionError: assert 'pem_path is required' in 'ERROR: pem_path (plaintext) or --cerberus-pem-gpg PATH (encrypted) is required unless --dry-run is set.\n'
1 failed in 0.11s
```

Fails the same way every run. No env var, no machine state involved — I initially suspected an environment dependency and was wrong; it's a plain stale assertion.

## How it was found

Surfaced during the #1378 investigation: a full-suite run in a clean worktree off `main` (`3c00c6465`) reported `1 failed, 5788 passed`. This is that one failure. It is unrelated to #1378 (inventory/worktree dirtying) — filed separately per one-issue-per-concern.

## The fix

Update the assertion in the test to match the current message. The robust form is to assert on the stable parts rather than a fragile exact phrase, e.g.:

```python
assert "pem_path" in err and "required" in err
```

or match the literal current wording. No production code change — the tool's behavior is correct as-is.

## Acceptance

- [ ] `test_main_returns_1_when_no_pem_and_no_dry_run` passes against current `main`.
- [ ] Assertion is resilient to the both-PEM-sources wording (doesn't depend on the exact phrase `pem_path is required`).
- [ ] No change to `tools/deploy_cerberus_secrets.py` (it is behaving correctly).
- [ ] Full suite returns 0 failures.

## Location

- Test: `tests/test_deploy_cerberus_secrets.py:129-133`
- Code (correct, do not change): `tools/deploy_cerberus_secrets.py:314-318`
- Likely-introducing change: #1254 (added `--cerberus-pem-gpg`, reworded the required-arg error)

--- a/tests/test_deploy_cerberus_secrets.py
+++ b/tests/test_deploy_cerberus_secrets.py
@@ -130,7 +130,7 @@ def test_main_returns_1_when_no_pem_and_no_dry_run(capsys):
     rc = deploy_cerberus_secrets.main([])  # neither pem nor --dry-run
     assert rc == 1
     err = capsys.readouterr().err
-    assert "pem_path is required" in err
+    assert "pem_path (plaintext) or --cerberus-pem-gpg PATH (encrypted) is required" in err
 
 
 # ---- #1118: dual-scope (Actions + Dependabot) deployment ----

